integrations/pinecone/tests/test_document_store_async.py

```diff
@@ -0,0 +1,156 @@
+import os
+import time
+
+import numpy as np
+import pytest
+from haystack import Document
+from haystack.components.preprocessors import DocumentSplitter
+from haystack.components.retrievers import SentenceWindowRetriever
+from pinecone import PineconeAsyncio
+
+from haystack_integrations.components.retrievers.pinecone import PineconeEmbeddingRetriever
+from haystack_integrations.document_stores.pinecone import PineconeDocumentStore
+
+
+@pytest.mark.integration
+@pytest.mark.asyncio
+@pytest.mark.skipif(not os.environ.get("PINECONE_API_KEY"), reason="PINECONE_API_KEY not set")
+async def test_serverless_index_creation_from_scratch(sleep_time):
```

Comment:
since this test is quite particular, takes time and is based on a fixed index name,
I think that it is not necessary for testing the async implementation and I would remove it.